src/sentry/search/snuba/backend.py

```diff
@@ -499,49 +494,53 @@ def snuba_search(start, end, project_ids, environment_ids, tags,
     if cursor is not None:
         having.append((sort_field, '>=' if cursor.is_prev else '<=', cursor.value))
 
-    # {group_id -> {<agg_alias> -> <agg_value>,
-    #               <agg_alias> -> <agg_value>,
-    #               ...},
-    #  ...}
-    # _OR_ if there's only one <agg_alias> in use
-    # {group_id -> <agg_value>,
-    #  ...}
-    snuba_results = snuba.query(
+    selected_columns = []
+    if get_sample:
+        # Get a random sample of matching groups. Because we use any(rand()),
+        # we are testing against a single random value per group, and so the
+        # sample is independent of the number of events in a group.
+        #
+        # The number of samples required to achieve a certain error bound with
+        # a certain confidence interval can be calculated from a rearrangement
+        # of the normal approximation (Wald) confidence interval formula:
+        #
+        # https://en.wikipedia.org/wiki/Binomial_proportion_confidence_interval
+        #
+        # Effectively if we want the estimate to be within +/- 10% of the real
+        # value with 95% confidence, we would need (1.96^2 * p*(1-p)) / 0.1^2
+        # samples. With a starting assumption of p=0.5 (this requires the most
+        # samples) we would need 96 samples to achieve +/-10% @ 95% confidence.
+
+        limit = options.get('snuba.search.hits-sample-size')
```

Comment:
Doesn't this shadow the existing `limit` argument? Should the return value be truncated to the (parameter) limit? What happens if the sample size is smaller? (Are we just assuming it won't be?)

Reply:
Reading further, seems like `get_sample` is only called in the case where we're pulling the sample (duh, I guess.) I think at least some of these arguments should probably be passed by the caller? I didn't look over this super thoroughly so I might be missing a reason why that's not practical.